src/main/java/sirius/db/es/types/DenseVectorProperty.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.db.es.types;
+
+import com.alibaba.fastjson.JSONArray;
+import com.alibaba.fastjson.JSONObject;
+import sirius.db.es.ESPropertyInfo;
+import sirius.db.mixing.AccessPath;
+import sirius.db.mixing.EntityDescriptor;
+import sirius.db.mixing.Mixing;
+import sirius.db.mixing.Property;
+import sirius.db.mixing.PropertyFactory;
+import sirius.kernel.commons.Value;
+import sirius.kernel.di.std.Register;
+import sirius.kernel.health.Exceptions;
+
+import javax.annotation.Nonnull;
+import java.lang.reflect.Field;
+import java.lang.reflect.Modifier;
+import java.util.function.Consumer;
+
+/**
+ * Properly handles {@link DenseVector} fields in {@linkplain sirius.db.es.ElasticEntity elastic entities}.
+ */
+public class DenseVectorProperty extends Property implements ESPropertyInfo {
+
+    /**
+     * Factory for generating properties based on their field type
+     */
+    @Register
+    public static class Factory implements PropertyFactory {
+
+        @Override
+        public boolean accepts(EntityDescriptor descriptor, Field field) {
+            return DenseVector.class.equals(field.getType());
+        }
+
+        @Override
+        public void create(EntityDescriptor descriptor,
+                           AccessPath accessPath,
+                           Field field,
+                           Consumer<Property> propertyConsumer) {
+            DenseVectorProperty denseVectorProperty = new DenseVectorProperty(descriptor, accessPath, field);
+
+            try {
+                if (field.get(accessPath.apply(descriptor.getType().getDeclaredConstructor().newInstance())) == null) {
+                    Mixing.LOG.WARN("Field %s in %s is a DenseVector. Such fields should be initialized as it"
+                                    + " should never be null!", field.getName(), field.getDeclaringClass().getName());
+                }
+
+                if (Modifier.isFinal(field.getModifiers())) {
+                    Mixing.LOG.WARN("Field %s in %s is a DenseVector and should be final!",
```

Comment:
Warning message does not match if-Statement. Shouldn't the boolean expression inside the if statement be negated?